js/modules/k6/experimental/fs/module.go

```diff
@@ -154,8 +157,14 @@ type File struct {
 	// Path holds the name of the file, as presented to [Open].
 	Path string `json:"path"`
 
-	// file contains the actual implementation for the file system.
-	file
+	// Impl contains the actual implementation of the file logic, and
+	// interacts with the underlying file system.
+	//
+	// Note that we explicitly omit exposing this to JS to avoid leaking
+	// implementation details, but keep it public so that we can access it
+	// from other modules that would want to leverage its implementation of
+	// io.Reader and io.Seeker.
+	Impl file `js:"-"`
```

Comment:
Nit; I think in general it's discouraged to expose an attribute as part of the public surface of the package (`fs` in this case) which type is private.

Reply:
I agree, this is an abstraction leak. The reason I went with that as a start was because I found the `Read`/`ExportedRead` approach we had initially taken somewhat inelegant (at least it tickled my sense of "nice" code).

But in hindsight it's probably the best compromise indeed.

The `Impl` approach that's explicitly not exposed to users, but to other k6 libs and modules, started from the rationale that the underlying `file` struct already implements `ReadSeeker`, and I might as well leverage that. But because I couldn't find a way to retrieve the underlying `file` implementation struct using `ExportTo` or by doing it manually, I went with a public `Impl` instead.

I'll give another shot at our initial approach and try to make it look okay 👍🏻

Reply:
Would it be feasible to define it as a `io.ReadSeeker`? 🤔  

If possible, even if that requires a couple of tricks internally (because of `stat()`), maybe that's a decent way to expose it for other k6 libs and modules.

Reply:
Do you mean?

```go
type File struct {
    // ...
    
	ReadSeeker io.ReadSeeker `js:"-"`
	
	// ...
}
```

If so, it's something I considered, and I think it could work (would have to double-check) 🙇‍♂️

Reply:
Something like that, yeah. I think it won't work as-is, because of `stat()`, as I said, but perhaps we can find some workaround that only lives internally/privately in the package/implementation, with the benefit of keeping the public surface cleaner.

Reply:
Let me see what I can do 🙇‍♂️

Reply:
Alright, I've pushed a [commit](https://github.com/grafana/k6/pull/3743/commits/b8def4c0bbcd539782dc63a65acb83f23159ff8f), which exposes the behavior as an interface instead. I like it much more, indeed.

I've created a `Stater` interface, made `file` implement it, and incorporated it in a `ReadSeekStater` interface, which I use to expose the underlying behavior instead of `file` itself. Let me know what you think 👍🏻

Reply:
It looks much cleaner now, at the very least! 👌🏻